nbs/models.ipynb

```diff
@@ -83,6 +83,7 @@
     "\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #48.            reconciliation = config.pop('reconciliation')
</pre><p>It isn't possible to do this check with optuna, so this model doesn't suport it. Can you please add a check for the backend in the init and if it's not ray raise an error? Or if you can make it work that'd be awesome, I just couldn't think of a way to remove this key from the function body.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/927/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='97'/>

Reply:
I think this boils down to essentially the same issue as your other remark: Optuna just doesn't work with Multivariate models and AutoHINT, as we need to enforce some parameters in the config. Let me think what would be best to solve it.

Reply:
I included an exception for now, I haven't found a way to get Optuna to work, it fails in base_auto on `metrics = trainer.callback_metrics`, which is `None` when using the redefined `_fit_model` of `AutoHINT` (and haven't figured out how to correct this.)

Reply:
Oh that should be easy to fix, that's the second argument of the return from _fit_model in AutoHINT, so you should go to HINT and store the trainer that is returned from the self.model.fit and propagate that back. I know it's hideous haha but I'm already working on improving it.